url0bbb975254_maxime_collin_mediatheque_tgz-pJ8-test_MainJ8/src/controleurs/ControleurUser.java

```diff
@@ -0,0 +1,174 @@
+package controleurs;
+
+import scolarite.Cours;
+import test.Bdd;
+import utilisateurs.Utilisateur;
+
+public class ControleurUser {
+	
+	ControleurEtudiant controleurEtudiant = new ControleurEtudiant();
+	ControleurEnseignant controleurEnseignant = new ControleurEnseignant();
+
+	public Boolean addUser(int idUser, String n, String p, String t, int nbEM) {
+		if (t == "Etudiant")
+			return controleurEtudiant.addEtudiant(idUser, n, p, t, nbEM);
+		
+		if (t == "Enseignant")
+			return controleurEnseignant.addEnseignant(idUser, n, p, t, nbEM);
+		
+		// si le type n'est ni Etudiant, ni Enseignant : echec
+		return false;
+	}
+
+	public Boolean setUser(int idUser, String n, String p, String t, int nbEM) {
+		if (t == "Etudiant")
+			return controleurEtudiant.setEtudiant(idUser, n, p, t, nbEM);
+		
+		if (t == "Enseignant")
+			return controleurEnseignant.setEnseignant(idUser, n, p, t, nbEM);
+		
+		// si le type n'est ni Etudiant, ni Enseignant : echec
+		return false;
+	}
+
+	public Boolean removeUser(int idUser) {
+		// on cherche dans la liste des Etudiants si idUser existe
+		int i = 0;
+		Boolean found = false;
+		while (i < Bdd.listEtudiants.size() && !found) {
+			if (Bdd.listEtudiants.get(i).getId() == idUser)
+				found = true;
+			else
+				i++;
+		}
+		
+		// si oui, on supprime l'etudiant
+		if (found) {
+			desinscrireUserAllCours(Bdd.listEtudiants.get(i));
+			Bdd.listEtudiants.remove(i);
+			return true;
+
+		// sinon on cherche dans la liste des enseignants si idUser existe 
+		} else {
+			i = 0;
+			while (i < Bdd.listEnseignants.size() && !found) {
+				if (Bdd.listEnseignants.get(i).getId() == idUser)
+					found = true;
+				else
+					i++;
+			}
+			
+			// si oui on supprime l'enseignant
+			if (found) {
+				System.out.println(Bdd.listEnseignants.get(i));
+				desinscrireUserAllCours(Bdd.listEnseignants.get(i));
+				Bdd.listEnseignants.remove(i);
+				return true;
+			}
+
+			// sinon : echec
+			else
+				return false;
+		}
+	}
+
+	private void desinscrireUserAllCours(Utilisateur user) {
+		// pour desinscrire un utilisateur de tous ses cours
+		// il faut recuperer chacun de ses cours
+		// le supprimer dans le cours
+		// puis supprimer tous ses cours
+		
+		// pour chacun de ses cours
+		int indexCoursDeUser = 0;
+		while (indexCoursDeUser < user.getCours().size()) {
+			// on recupere le cours
+			Cours cours = user.getCours().get(indexCoursDeUser);
+			
+			if (user.getType() == "Etudiant")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.